### PR TITLE
SRE-396 processing windows eventlog events

### PR DESCRIPTION
**Pull request by czm41k**
1. HTTP input for processing WindowsEvents
2. WinEvent processor: implemented both HandleEvent for pubsub & HandleHttpRequest for HTTP input
3. Sample of winevent http json payload added to tests/
4. WinEvent in Grafana output templates

### CODE REVIEW

File: processor/winevent.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	p.requests.Inc(channel)
+
+	var body []byte
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	// This snippet stands to immitate pubsub input
```
Comment (XCiber) [inline processor/winevent.go:104]: pls remove commented code

File: processor/winevent.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	Original *WinEventOriginalRequest `json:"data"`
+}
+type WinEventOriginalRequest struct {
+	Events []WinEvent `json:"metrics"`
+}
+type WinEventTags struct {
+	Message      string `json:"Message"`
+	MessageLevel string `json:"LevelText"`
+	Keywords     string `json:"Keywords"`
+	EventId      string `json:"EventRecordID"`
```
Comment (XCiber) [inline processor/winevent.go:28]: should be EventID by go naming convention

File: processor/winevent.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	return nil
+}
+
+func (p *WinEventProcessor) send(span sreCommon.TracerSpan, channel string, event interface{}, t *time.Time) {
+	e := &common.Event{
+		Channel: channel,
+		Type:    p.EventType(),
+		Data:    event,
+	}
+	if t != nil && (*t).UnixNano() > 0 {
```
Comment (XCiber) [inline processor/winevent.go:166]: could simplify (*t).UnixNano to t.UnixNano

File: processor/winevent.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+		e.SetTime((*t).UTC())
```
Comment (XCiber) [inline processor/winevent.go:167]: could simplify (*t).UTC to t.UTC

**Review by XCiber — CHANGES_REQUESTED** [review]
LGTM
Few small hints

File: processor/winevent.go (hunk 2 above)
Comment (tsvtitan) [reply processor/winevent.go:28]: This should use golang notation. Agree

File: processor/winevent.go (hunk 1 above)
Comment (tsvtitan) [reply processor/winevent.go:104]: Agree

**Pull request merged**